r/tests/testthat/test-dplyr-collect.R

```diff
@@ -276,3 +276,16 @@ test_that("collapse doesn't unnecessarily add ProjectNodes", {
   )
   expect_length(grep("ProjectNode", plan), 1)
 })
+
+test_that("compute", {
```

Comment:
I would put this test in `test-dplyr-query.R` and rename this file back to `test-dplyr-collapse.R` since it's all about `collapse()`. I get the point of trying to have the R and test file names matching, but it's not so straightforward here--nearly every dplyr test we have calls `collect()` so we're testing it everywhere.

Also let's give the test a more expressive name

```suggestion
test_that("compute() on a grouped query returns a Table with groups in metadata", {
```